client_code/routing/__init__.py

```diff
@@ -7,25 +7,194 @@
 
 __version__ = "1.9.0"
 
-from ._routing import (
-    add_to_cache,
-    clear_cache,
-    error_form,
-    get_cache,
+from anvil.js import window as _w
+
+from . import _navigation
+from . import _router as _r
+from ._decorators import error_form, route, template
+from ._logging import log as _log
```

Comment:
I can't see a `log` attribute in the `._logging` module. What's this importing?

Reply:
It should be in the diff at the end of the _logging module

It's just a function to send a callable to logger.print, which just means we don't need to realise the f-string if we're not in debug mode. It's like a tiny performance win.

Reply:
ah. stupid of me not to check if it's something new in here!